mux/mux.go

```diff
@@ -19,10 +19,30 @@ type MuxedStream interface {
 	io.Reader
 	io.Writer
 
-	// Close closes the stream for writing. Reading will still work (that
-	// is, the remote side can still write).
+	// Close closes the stream.
+	//
+	// * Any buffered data for writing will be flushed.
```

Comment:
Hum, this is **abnormal** termination. In TCP sockets, a `close()` queues a `FIN`, and honours reliability until that `FIN` is acked. In other words, discarding all queued data turns this operation in to a `Reset()`, no?

Reply:
This is normal termination. All buffered data for writing will be _flushed_, not discarded. Unread data will be discarded but that's no different from POSIX (where the file descriptor itself will be closed).

Reply:
Duh, I misread 🤷‍♂